invenio_app_ils/ui/src/invenio_app_ils/authentication/components/AuthenticationGuard/AuthenticationGuard.js

```diff
@@ -0,0 +1,36 @@
+import React, { Component } from 'react';
+import { Redirect } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import {
+  sessionManager,
+  authenticationService,
+} from '../../../authentication/services';
+
+export class AuthenticationGuard extends Component {
+  render() {
+    let { component: Component, roles, ...restParams } = this.props;
+
+    if (!sessionManager.isUserLoggedIn()) {
+      authenticationService.login(window.location.pathname);
+      return null;
+    }
+
+    if (sessionManager.isUserLoggedIn() && !sessionManager.hasRoles(roles)) {
+      // TODO: Handle the insufficient permissions for user
+      console.error(
+        `User has no permission to access the page ${window.location.pathname}`
+      );
+      return <Redirect to={'/'} />;
```

Comment:
Instead of hardcoding urls, we can create a file similar to `BackofficeURLs`, or even merge the url files and have something like `<Redirect to={urls.base}>`